### PR TITLE
App Gateway docs QS template

### PR DESCRIPTION
**Pull request by vhorne**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by vhorne** [comment]
#sign-off

File: ag-docs-qs/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('networkInterfaces_myvm584_name'))]"
+                        }
+                    ]
+                }
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline ag-docs-qs/azuredeploy.json]: use a copy loop here as well - i.e. for all your VM resources

File: ag-docs-qs/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+                "autoUpgradeMinorVersion": true,
+                "publisher": "Microsoft.Compute",
+                "type": "CustomScriptExtension",
+                "typeHandlerVersion": "1.4",
+                "settings": {
+                    "commandToExecute": "powershell Add-WindowsFeature Web-Server; powershell Add-Content -Path \"C:\\inetpub\\wwwroot\\Default.htm\" -Value $($env:computername)"
+                }
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline ag-docs-qs/azuredeploy.json]: why are we deploying the NSG rules twice?

File: ag-docs-qs/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+                "sourcePortRange": "*",
+                "destinationPortRange": "3389",
+                "sourceAddressPrefix": "*",
+                "destinationAddressPrefix": "*",
+                "access": "Allow",
+                "priority": 300,
+                "direction": "Inbound"
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline ag-docs-qs/azuredeploy.json]: why are we deploying the subnets twice?

File: ag-docs-qs/azuredeploy.parameters.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "value": "GEN-UNIQUE"
+    },
+    "adminPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "location": {
```
Comment (bmoore-msft) [inline ag-docs-qs/azuredeploy.parameters.json]: why centralus?

File: ag-docs-qs/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,548 @@
...
+            ],
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('vmSize')]"
+                },
+                "storageProfile": {
+                    "imageReference": {
+                        "publisher": "MicrosoftWindowsServer",
+                        "offer": "WindowsServer",
+                        "sku": "2016-Datacenter",
```
[the remark this answers is not in the document]
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: I want to keep this consistent with the other Application Gateway Quickstarts (PS, CLI, portal) that use 2016.

File: ag-docs-qs/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,548 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "String",
```
[the remark this answers is not in the document]
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: fixed

File: ag-docs-qs/azuredeploy.json (hunk 2 above)
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: Removed nsg securityRules resources

File: ag-docs-qs/azuredeploy.parameters.json (hunk 4 above)
Comment (vhorne) [reply ag-docs-qs/azuredeploy.parameters.json]: No particular reason. Removed

File: ag-docs-qs/README.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# Azure Application Gateway v2 Quickstart
+
+<IMG SRC="https://azurequickstartsservice.blob.core.windows.net/badges/100-blank-template/PublicLastTestDate.svg" />&nbsp;
```
[the remark this answers is not in the document]
Comment (vhorne) [reply ag-docs-qs/README.md]: Copied the markdown from the validation output into my Readme. There weren't any instructions about what might changes are needed, so hopefully the markdown listed in the Validation is what is needed.

File: ag-docs-qs/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,548 @@
...
+                        "sku": "2016-Datacenter",
+                        "version": "latest"
+                    },
+                    "osDisk": {
+                        "osType": "Windows",
+                        "name": "[concat(variables('virtualMachines_myVM_name'), '_disk1_843fbb3acb4d4aa4bc0df6050ee2728f')]",
+                        "createOption": "FromImage",
+                        "caching": "ReadWrite",
+                        "managedDisk": {
+                            "storageAccountType": "Premium_LRS"
```
[the remark this answers is not in the document]
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: Changed to Standard SSD

File: ag-docs-qs/azuredeploy.json (hunk 3 above)
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: No idea. Azure portal created the template. Removed the subnet resources and adjusted the DependsOn statements to account for the change.

File: ag-docs-qs/azuredeploy.json (hunk 1 above)
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: done

File: ag-docs-qs/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,548 @@
...
+                            "destinationAddressPrefix": "*",
+                            "access": "Allow",
+                            "priority": 300,
+                            "direction": "Inbound"
+                        }
+                    }
+                ]
+            }
+        },
+        {
```
[the remark this answers is not in the document]
Comment (vhorne) [reply ag-docs-qs/azuredeploy.json]: done

**Pull request merged**